src/overlays/actors/ovl_En_Pamera/z_en_pamera.c

```diff
@@ -22,135 +104,791 @@ const ActorInit En_Pamera_InitVars = {
+
+    while (actor != NULL) {
+        if ((actor->id == ACTOR_EN_DOOR) && (Math_Vec3f_DistXZ(&this->actor.world.pos, &actor->world.pos) < 200.0f)) {
+            this->actor.child = actor;
+            return true;
+        }
+        actor = actor->next;
+    }
+    this->actor.child = NULL;
+    return false;
+}
+
+void func_80BD8588(EnPamera* this, GlobalContext* globalCtx) {
+    Path* path = &globalCtx->setupPathList[((this->actor.params & 0xFF0) >> 4)];
+    Vec3f sp28;
+
+    if (path == NULL) {
+        Actor_MarkForDeath(&this->actor);
+    }
+    if (gSaveContext.weekEventReg[61] & 4) {
+        path = &globalCtx->setupPathList[path->unk1];
+    }
+    this->pathPoints = Lib_SegmentedToVirtual(path->points);
+    this->pathIndex = 0;
+    this->pathPointsCount = path->count;
+    this->pathID = path->unk1;
+    Math_Vec3s_ToVec3f(&sp28, this->pathPoints);
+    this->actor.world.rot.y = this->actor.shape.rot.y = Math_Vec3f_Yaw(&this->actor.world.pos, &sp28);
+}
+
+void func_80BD8658(EnPamera* this) {
+    s32 i;
+    s16 nextCutscene = this->actor.cutscene;
+
+    for (i = 0; i < ARRAY_COUNT(this->cutscenes); i++) {
+        if (nextCutscene == -1) {
+            break;
+        }
+        this->cutscenes[i] = nextCutscene;
+        nextCutscene = ActorCutscene_GetAdditionalCutscene(nextCutscene);
+    }
+}
+
+void EnPamera_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnPamera* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_80BD8700(EnPamera* this) {
+    this->hideInisdeTimer = 0;
+    this->actor.flags &= ~1;
+    func_800BDC5C(&this->skelAnime, sAnimations, 0);
+    this->actionFunc = func_80BD8758;
+}
+
+void func_80BD8758(EnPamera* this, GlobalContext* globalCtx) {
+    if (this->hideInisdeTimer++ > 1800) {
+        if (ActorCutscene_GetCanPlayNext(this->cutscenes[0]) && (this->cutscenes[0] != -1)) {
+            ActorCutscene_StartAndSetUnkLinkFields(this->cutscenes[0], &this->actor);
+            func_800E02AC(Play_GetCamera(globalCtx, ActorCutscene_GetCurrentCamera(this->cutscenes[0])), this);
+            this->actor.speedXZ = 1.5f;
+            func_800BDC5C(&this->skelAnime, sAnimations, 1);
+            this->actor.shape.rot.y = this->actor.home.rot.y;
+            this->actor.world.rot.y = this->actor.home.rot.y;
+            func_80BD9338(this, globalCtx);
+            func_80BD8908(this);
+        } else {
+            if ((this->cutscenes[0] != -1) && (this->actor.xzDistToPlayer < 1000.0f)) {
+                ActorCutscene_SetIntentToPlay(this->cutscenes[0]);
+            } else {
+                this->actor.speedXZ = 1.5f;
+                func_800BDC5C(&this->skelAnime, sAnimations, 1);
+                this->actor.shape.rot.y = this->actor.home.rot.y;
+                this->actor.world.rot.y = this->actor.home.rot.y;
+                func_80BD9338(this, globalCtx);
+                func_80BD8908(this);
+            }
+        }
```

Comment:
You can mix those into one `else if` if you think it makes sense. Again, it is up to you.
```suggestion
        } else if ((this->cutscenes[0] != -1) && (this->actor.xzDistToPlayer < 1000.0f)) {
            ActorCutscene_SetIntentToPlay(this->cutscenes[0]);
        } else {
            this->actor.speedXZ = 1.5f;
            func_800BDC5C(&this->skelAnime, sAnimations, 1);
            this->actor.shape.rot.y = this->actor.home.rot.y;
            this->actor.world.rot.y = this->actor.home.rot.y;
            func_80BD9338(this, globalCtx);
            func_80BD8908(this);
        }
```